adapters/adapterstest/test_json.go

```diff
@@ -263,60 +279,61 @@ func diffBids(t *testing.T, description string, actual *adapters.TypedBid, expec
 		return
 	}
 
-	diffStrings(t, fmt.Sprintf("%s.type", description), string(actual.BidType), string(expected.Type))
-	diffOrtbBids(t, fmt.Sprintf("%s.bid", description), actual.Bid, expected.Bid)
+	assert.NoError(t, diffStrings(fmt.Sprintf("%s.type", description), string(actual.BidType), string(expected.Type)))
+	assert.NoError(t, diffOrtbBids(fmt.Sprintf("%s.bid", description), actual.Bid, expected.Bid))
 }
 
 // diffOrtbBids compares the actual Bid made by the adapter to the expectation from the JSON file.
-func diffOrtbBids(t *testing.T, description string, actual *openrtb2.Bid, expected json.RawMessage) {
+func diffOrtbBids(description string, actual *openrtb2.Bid, expected json.RawMessage) error {
 	if actual == nil {
-		t.Errorf("Bidders cannot return nil Bids. %s was nil.", description)
-		return
+		return fmt.Errorf("Bidders cannot return nil Bids. %s was nil.", description)
 	}
 
 	actualJson, err := json.Marshal(actual)
 	if err != nil {
-		t.Fatalf("%s failed to marshal actual Bid into JSON. %v", description, err)
+		return fmt.Errorf("%s failed to marshal actual Bid into JSON. %v", description, err)
 	}
 
-	diffJson(t, description, actualJson, expected)
+	return diffJson(description, actualJson, expected)
 }
 
-func diffStrings(t *testing.T, description string, actual string, expected string) {
+func diffStrings(description string, actual string, expected string) error {
 	if actual != expected {
-		t.Errorf(`%s "%s" does not match expected "%s."`, description, actual, expected)
+		return fmt.Errorf(`%s "%s" does not match expected "%s."`, description, actual, expected)
 	}
+	return nil
 }
 
 // diffJson compares two JSON byte arrays for structural equality. It will produce an error if either
 // byte array is not actually JSON.
-func diffJson(t *testing.T, description string, actual []byte, expected []byte) {
+func diffJson(description string, actual []byte, expected []byte) error {
 	if len(actual) == 0 && len(expected) == 0 {
-		return
+		return nil
 	}
 	if len(actual) == 0 || len(expected) == 0 {
-		t.Fatalf("%s json diff failed. Expected %d bytes in body, but got %d.", description, len(expected), len(actual))
+		return fmt.Errorf("%s json diff failed. Expected %d bytes in body, but got %d.", description, len(expected), len(actual))
```

Comment:
Now that `diffJson()` returns an error, no need to differentiate this error as`Fatalf` over `Errorf`. This particular JSON test won't continue to run because our test framework will simply go to the next JSON file in the directory.